clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -193,7 +448,21 @@ LogicalResult LowerFunction::buildFunctionEpilog(const LowerFunctionInfo &FI) {
         return success();
       }
     } else {
-      llvm_unreachable("NYI");
+      // NOTE(cir): Unlike the original codegen, CIR may have multiple return
+      // statements in the function body. We have to handle this here.
+      mlir::PatternRewriter::InsertionGuard guard(rewriter);
+      NewFn->walk([&](ReturnOp returnOp) {
+        rewriter.setInsertionPoint(returnOp);
+
+        // TODO(cir): I'm not sure if we need this offset here or in CIRGen.
+        // Perhaps both? For now I'm just ignoring it.
```

Comment:
Did you try adding an assert on the OG path and running `ninja check-clang-codegen`? That might give you a testcase where this path is used, and can help you figure out the answer here.